### ISSUE TITLE
Browser Select window on top

### ISSUE THREAD

**Issue by ducmthai** [zumoshi/BrowserSelect#59]
Hi,

Currently, when a link is clicked in Office apps and Windows 10 settings send to Browser Select but its window is not in foreground or on top of other program windows. Can we have an option to make it always on top so we don't have to minimize window after window in order to see Browser Select?

Thanks,
Jasper

**Comment by octogonz** [comment]
@zumoshi I'm willing to try implementing this.  Would you accept such a PR?

**Comment by zumoshi** [comment]
Sure, I'm always happy to accept PR's. Just make it an opt-in option so it's not enabled by default.

**Comment by octogonz** [comment]
@ducmthai Do you remember how to repro this problem?  After I upgraded to 1.4.1 it seems to not be happening any more. But it only happened in certain situations -- maybe I'm just not trying the right thing. (?)

**Comment by ducmthai** [comment]
@octogonz Firstly, thanks for jumping in :)
Secondly, you can reproduce this behavior with most if not all UWP app. I click a link in Skype UWP or from Settings app BrowserSelect would open but not show up.

**Comment by octogonz** [comment]
I just tested with UWP **Skype**, **OneNote**, and **Microsoft News**. In all cases it seems to correctly switch to the BrowserSelect dialog when a hyperlink is clicked.

@ducmthai Are you using version 1.4.1?

**Comment by ducmthai** [comment]
Yes, I do. I also using Windows Insider slow ring. Don't know if that is the cause?

**Comment by Luckz** [comment]
For me clicking external links in Steam will in 99% of cases put BrowserSelect into the background, with no taskbar icon.

**Comment by octogonz** [comment]
@Luckz Which link are you clicking?  I tried clicking this:

![Untitled](https://user-images.githubusercontent.com/4673363/77838130-be615580-7125-11ea-9e0d-d0df261268b6.png)

...and the BrowserSelect.exe window correctly opens in front of the Steam window, with a taskbar icon.  In Process Explorer, I see a process tree like this:

![Untitled2](https://user-images.githubusercontent.com/4673363/77838162-18621b00-7126-11ea-9ad7-9c841eb922d5.png)

I'm using Windows 10 Pro 64-bit build 18363.70.  And BrowserSelect is version 1.4.1.

**Comment by Luckz** [comment]
![image](https://user-images.githubusercontent.com/224748/77838570-69e5c680-7175-11ea-8b2a-2da00e6481e4.png)

For me it just does not appear in the taskbar; build 18363.592 (and Browser Select 1.4.1).
I would assume it's not due to Steam Beta participation:
![image](https://user-images.githubusercontent.com/224748/77838582-8da90c80-7175-11ea-9468-f5b61f657875.png)


This works fine if I instead change my default browser to Browser Chooser 2:
![image](https://user-images.githubusercontent.com/224748/77838621-f7c1b180-7175-11ea-90f4-e9a771a81df4.png)
![image](https://user-images.githubusercontent.com/224748/77838627-10ca6280-7176-11ea-8d35-0f1c991fc0ad.png)

**Comment by octogonz** [comment]
🤔 Hrmm... Unfortunately I don't have time to install a VM with the newer Windows Insider release (at least, not unless we can conclusively prove that it is the cause).

Has anyone else encountered this with regular Windows 10?  I'll try to do some more experiments, or try it on some other computers.  We need a consistent repro before I can debug it.

**Comment by ducmthai** [comment]
Hi,

I did test it with another laptop running regular Windows 10 and BrowserSelect did pop-up and has a taskbar icon. However, on Windows 10 Insider Slow Ring it behaves like the image below:
![image](https://user-images.githubusercontent.com/13193688/77862533-b0710a80-71e1-11ea-825a-91407ab2f872.png)

**Comment by adynis** [comment]
I also get this issue, I didn't investigated the conditions, but I click a link, and ... I learned that I need to search the window somewhere in the background behind my windows ...

**Comment by ducmthai** [comment]
Is there any update on this issue? If you click a link in an UWP app you will see this behavior.

**Comment by gtflip** [comment]
I have a case where it pops up in the background, and there is no icon in the taskbar.  Once I switch to it using alt-tab, the taskbar icon shows up.

I use the npm-dependency-links extension in VS code, which lets you ctrl-click on a package name in package.json to open the npmjs.com page in your browser.  I'm pretty sure it doesn't pop in front since I'm holding a key down (I've seen that before on Windows), but the taskbar thing is more annoying.  Not a critical bug, obviously, just an annoyance.  😃

**Comment by lucasnz** [comment]
> I have a case where it pops up in the background, and there is no icon in the taskbar. Once I switch to it using alt-tab, the taskbar icon shows up.
> 
> I use the npm-dependency-links extension in VS code, which lets you ctrl-click on a package name in package.json to open the npmjs.com page in your browser. I'm pretty sure it doesn't pop in front since I'm holding a key down (I've seen that before on Windows), but the taskbar thing is more annoying. Not a critical bug, obviously, just an annoyance. 😃

Out of curiosity, do you have the same issue with my fork? https://github.com/lucasnz/BrowserSelect

I have included this PR, which in theory should resolve this issue: https://github.com/zumoshi/BrowserSelect/pull/80

You could also try the PR authors fork too...
